### PR TITLE
Fix Dune to bootstrap when the path to OCaml contains spaces 

### PR DESCRIPTION
**Pull request by Chris00**
See [build with this patch](https://ci.appveyor.com/project/Chris00/ocaml-appveyor/builds/26782076).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -121,6 +121,7 @@ If you do not have `make`, you can do the following:
 ```sh
 $ ocaml bootstrap.ml
 $ ./boot.exe
+$ ./_boot/default/bin/main.exe build @install
```
Comment (rgrinberg) [inline README.md]: I think we need the `--release` and `--subst` flags here as well. I general, I'd prefer not to deviate from the commands in the opam file btw. Perhaps we can just copy those and add an install step?

Comment (diml) [reply README.md]: It is not necessary to call both `boot.exe` and `main.exe`. Adding `--release` to the `boot.exe` call should be enough.

BTW, the `subst` call is no longer necessary now that we have `dune-build-info`. In particular, the `%%VERSION%%` string no longer appears in the code base, except in a comment.

Comment (Chris00) [reply README.md]: Using [boot --release](https://github.com/Chris00/ocaml-appveyor/commit/3e0ef47181f0278ea685c8484ffdec043ffe0a92#diff-e612dee15b2c34817a46aff4334a02fdR85) does [not work](https://ci.appveyor.com/project/Chris00/ocaml-appveyor/builds/26791368#L5913).

Comment (diml) [reply README.md]: Ah, the install command on the following line should take `--build-dir _boot`. The readme wasn't updated when we change the build directory during bootstrap

**Comment by aalekseyev** [comment]
TIL `Filename.quote_command` is [a thing](https://github.com/ocaml/ocaml/pull/1492) which should be used for this. It's only coming in OCaml 4.10 though.

**Comment by Chris00** [comment]
@aalekseyev I agree — when 4.10 will be the lower OCaml version Dune accepts!

**Comment by diml** [comment]
Let's merge this in the meantime since it helps. I rebased and merged by hand in master.

**Pull request closed without merging**